dipy/core/wavelet.py

```diff
@@ -0,0 +1,258 @@
+import numpy as np
+from dipy.denoise import nlmeans_block
+
+"""
+ Functions for Wavelet Transforms in 3D domain
+
+ Code adapted from
+
+ WAVELET SOFTWARE AT POLYTECHNIC UNIVERSITY, BROOKLYN, NY
+ http://taco.poly.edu/WaveletSoftware/
+"""
+
+
+def cshift3D(x, m, d):
+    """3D Circular Shift
+
+    Parameters
+    ----------
+    x: N1 by N2 by N3 array
+    m: amount of shift
+    d: dimension of shift (d = 1,2,3)
+
+    Returns
+    -------
+    y: array x will be shifed by m samples down
+       along dimension d
+
+    """
+    s = x.shape
+    idx = (np.array(range(s[d])) + (s[d] - m % s[d])) % s[d]
+    idx = np.array(idx, dtype=np.int64)
+    if d == 0:
+        return x[idx, :, :]
+    elif d == 1:
+        return x[:, idx, :]
+    else:
+        return x[:, :, idx]
+
+
+def permutationinverse(perm):
+    """
+    Function generating inverse of the permutation
+    """
+
+    inverse = [0] * len(perm)
+    for i, p in enumerate(perm):
+        inverse[p] = i
+    return inverse
+
+
+def afb3D_A(x, af, d):
+    """3D Analysis Filter Bank
+     (along one dimension only)
+
+    Parameters
+    ----------
+    x:  N1xN2xN2 matrix, where min(N1,N2,N3) > 2*length(filter)
```

Comment:
Say here
```
x : array, size N1 x N2 x N3
    where min(N1, N2, N3) > 2 * length(filter)
```

And add a space after every input parameter in the docstring

```
af:
```
should be
```
af :
```